adapters/rest-adapter-generator/src/main/java/uk/gov/justice/services/adapters/rest/generator/RestAdapterGenerator.java

```diff
@@ -104,15 +66,16 @@ private void writeToBasePackage(final TypeSpec typeSpec, final GeneratorConfig c
      */
     private List<String> writeToBasePackage(final List<TypeSpec> typeSpecs,
                                             final GeneratorConfig configuration,
-                                            final String packageName) {
+                                            final String subPackageName) {
 
-        final List<String> implementationNames = new ArrayList<>();
+        final List<String> implementationNames = new LinkedList<>();
+        final String packageName = packageNameOf(configuration, subPackageName);
 
         typeSpecs.stream()
                 .forEach(typeSpec -> {
                     try {
                         if (classDoesNotExist(configuration, typeSpec)) {
-                            JavaFile.builder(configuration.getBasePackageName() + packageName, typeSpec)
+                            JavaFile.builder(packageName, typeSpec)
```

Comment:
If made static it just says builder and does not give meaning to what is happening, I prefer to keep as is.